src/main/java/com/fasterxml/jackson/core/JsonParser.java

```diff
@@ -437,6 +439,26 @@ protected JsonParser() {
      */
     public abstract void setCodec(ObjectCodec oc);
 
+    /**
+     * Set the max number of characters that are allowed when parsing numbers.
+     *
+     * @param maxNumLen max number of characters that are allowed
+     * @since 2.14
+     */
+    public void setMaxNumLen(int maxNumLen) {
```

Comment:
Quick comment: I would want to avoid setters (and probably getters too) since 3.0 has parser and generators fully immutable. Also settings should probably be passed in a separate config object to reduce number of methods/parameters being passed: something like `StreamReadConfig` / `StreamWriteConfig` (maybe), sub-classable on per-format basis. Configured using builder()-style approach; starting with default settings.

Reply:
Hmmh. Ok, first things first: I would suggest going via 2.15, not `master`, due to package renaming. It'll probably be tricky either way, but I don't think master->2.15 is any easier than the other way around.

But be that as it may, bigger question is that of passing config objects. My first gut reaction is that making these go through `IOContext` is not clean. But the thing about `TextBuffer` might be relevant -- passing to `JsonParser` / `JsonGenerator` I wouldn't worry as much about (factories have been changed a few times and while unfortunate it's internal interface).
I wouldn't be totally against passing it via `IOContext` but would prefer alternative.

One thing I do not remember, which may be relevant, is whether format backends sub-class `IOContext` or not; I have a vague recollection might do, for recycling or such.

Reply:
I've introduced a StreamReadConfig - and it can only be set via the TSFBuilder.

Internally, I still haven't found a better way to pass around the StreamReadConfig than using the IOContext.

Reply:
Apologies for not thinking this through. As per my CR comments, I think it should be `StreamReadLimits` (or perhaps `StreamReadConstraints`).

But one thing that is also needed is to make sure values are fully immutable so they can be shared by parser instances etc. Changes could be done using `wihtMaxNumberLength()` or such, but that may become unmaintainable when adding values. So perhaps it is necessary to start with builder style construction immediately.

However.... I guess I am ok without doing that right away so we could merge things in sooner.

So it is ok not to add builder as the first thing.

Otherwise I like this; passing limits to/via `IOContext` makes sense, I think.